server/AuthController.js

```diff
@@ -12,6 +13,42 @@ module.exports = {
     res.set(JWT.renewTokenHeader, token);
     res.status(200).send({ token })
   },
+  impersonate: (_grabUser, id='id') => async (req, res) => {
+    try {
+      let token = JWT.issue({
+        // The new user will be whatever the target id is
+        user: req.param(id),
```

Comment:
`sails-async` will let you just look at the third param (which is the result of `req.allParams()`) instead of `req.param`.